draft-bbs-signatures.md

```diff
@@ -841,6 +845,132 @@ Procedure:
 4. return (a_octets || e_octets || s_octets)
 ```
 
+### OctetsToProof
+
+The `octets_to_proof` operations describes how to parse an octet string representation of a proof, validate it and return the underlying components.
+
+The proof value outputted by this operation consists of the following components, in that order:
+
+1. Three (3) valid points of the G1 subgroup, different from the identity point.
+2. Five (5) integers representing scalars in the range of 1 to r-1 inclusive.
+3. A number of integers representing scalars in the range of 1 to r-1 inclusive, corresponding to the un-revealed from the proof message commitments.
+
+```
+proof = octets_to_proof(proof_octets)
+
+Inputs:
+
+- proof_octets (REQUIRED), octet string of the form output from
+                           proof_to_octets operation.
+
+Parameters:
+
+- r (REQUIRED), non-negative integer. The prime order of the G_1 and
+                G_2 groups, defined by the ciphersuite.
+- octet_scalar_length (REQUIRED), non-negative integer. The length of
+                                  a scalar octet representation, defined
+                                  by the ciphersuite.
+- octet_point_length (REQUIRED), non-negative integer. The length of
+                                 a point in G1 octet representation,
+                                 defined by the ciphersuite.
+
+Outputs:
+
+- proof, a proof value in the form described above or INVALID
+
+Procedure:
+
+1. proof_len_floor = 3 * octet_point_length + 5 * octet_scalar_length
+
+2. if length(proof_octets) < proof_len_floor, return INVALID
+
+// Points (i.e., (A', Abar, D) in ProofGen) de-serialization.
+3. index = 0
+
+4. for i in (0, 2):
```

Comment:
Just checking this loop syntax is 0,1,2 wonder if we should do `(0,..,2)` and define the syntax in the terminology section?

Reply:
Yeap! Done!